xbmc/video/VideoDatabase.h

```diff
@@ -592,19 +592,20 @@ class CVideoDatabase : public CDatabase
                            const std::map<std::string, std::string>& artwork,
                            int idShow,
                            int idEpisode = -1);
-  bool SetFileForEpisode(const std::string& fileAndPath, int idEpisode, int idFile);
-  bool SetFileForMovie(const std::string& fileAndPath, int idMovie, int idFile);
+  bool SetFileForEpisode(const std::string& fileAndPath, const int idEpisode, const int oldIdFile);
+  bool SetFileForMovie(const std::string& fileAndPath, const int oldIdFile);
   int SetDetailsForMusicVideo(CVideoInfoTag& details,
                               const std::map<std::string, std::string>& artwork,
                               int idMVideo = -1);
-  int SetStreamDetailsForFile(const CStreamDetails& details, const std::string& strFileNameAndPath);
+  bool SetStreamDetailsForFile(const CStreamDetails& details,
+                               const std::string& strFileNameAndPath);
   /*!
    * \brief Clear any existing stream details and add the new provided details to a file.
    * \param[in] details New stream details
    * \param[in] idFile Identifier of the file
    * \return operation success. true for success, false for failure
    */
-  bool SetStreamDetailsForFileId(const CStreamDetails& details, int idFile);
+  bool SetStreamDetailsForFileId(const CStreamDetails& details, int idFile) const;
```

Comment:
A setter with a const modifier is a red flag. Setters clearly mutate state, so shouldn't be const. And as long as I noticed this, const should be dropped from the DB getters too. The getters call into mysql, and even though it's a DB read, it still mutates internal state of the sql object.